analyzers/src/SonarAnalyzer.CSharp/SymbolicExecution/Roslyn/PublicMethodArgumentsShouldBeCheckedForNull.cs

```diff
@@ -29,5 +29,81 @@ public class PublicMethodArgumentsShouldBeCheckedForNull : SymbolicRuleCheck
 
     protected override DiagnosticDescriptor Rule => S3900;
 
-    public override bool ShouldExecute() => false;
+    public override bool ShouldExecute()
```

Comment:
@pavel-mikula-sonarsource This method is so complicated that it should be under test. What do you think about this:
* Add a method `internal static bool ShouldExecuteImpl(SyntaxNode node)` and ShouldExecute() delegates there.
* Add tests like so:
```cs
[TestMethod]
public void Test()
{
    var (tree, model) = TestHelper.CompileCS("some test code");
    var node = FindTheNode(tree);
    var actual = PublicMethodArgumentsShouldBeCheckedForNull.ShouldExecuteImpl(node);
    actual.Should().BeTrue();
}
```
This is a typical pattern to test stateless/pure functions of a class and should be easy to do. By making ShouldExecuteImpl static, we also ensure that we can not query the semantic model or do anything other than syntax-based checks. (Which in this case is a problem: If the method only has non-nullable value types as parameters, ShouldExecute should return false, but we can not check that).

Reply:
As discussed earlier, I'll incorporate these into the test cases by adding Compliant scenarios when `ShouldExecute()` returns `false`. I'll do this once I have the rule implementation in a separate PR.